README.md

```diff
@@ -0,0 +1,141 @@
+# Setting up the API Endpoints
+
+This README provides instructions for setting up the database, configuring users, and connecting the client and server in development mode.
+
+First, clone the repository and navigate to the project directory.
+
+```bash
+git clone org-184059057@github.com:hellowhorl/whorl-server.git
+```
+
+## Setup a Virtual Environment
+
+Here you will utilize `venv`, which is short for virtual environment, specific for your client.
+
+1. **Create a Virtual Environment**  
+    - In the root of the `whorl-server` repository, run the following command:
+
+      ```bash
+      python -m venv .venv
+      ```
+
+2. **Activate the Virtual Environment**  
+    - Activate your virtual environment with the command:
+
+      ```bash
+      source .venv/bin/activate
+      ```
+
+## Development Install
+
+Once your virtual environment is set up, perform a development install by running the following command:
+
+```bash
+python -m pip install -e .
+```
+
+## Keys and Environment Variables
+
+If you are part of the `whorl` organization, you can find the keys in the `.env` file. If you are not, you can create your own `.env` file in the root of the `whorl-server` repository.
+
+- The `.env` file should contain the following variables:
+
+     ```plaintext
+      API_URL=localhost
+      API_DB_HOST=localhost
+      API_DB_USER=<your_api_db_user_name>
+      API_DB_PASS=<your_api_db_pass>
+      API_HOST=<your_api_host>
+
+      OPEN_AI_KEY=<your_openai_key>
+
+      OPENWEATHER_API=<your_openweather_key>
+      OPENWEATHER_LAT=<your_openweather_latitude>
+      OPENWEATHER_LON=<your_openweather_longitude>
+     ```
+
+## PostgreSQL Setup
+
+1. **Install PostgreSQL**  
+   - Make sure PostgreSQL is installed on your machine.  
+   - If not installed, you can download it from [PostgreSQL Downloads](https://www.postgresql.org/download/) and follow the installation instructions for your operating system.
+
+2. **Access the PostgreSQL Shell**  
+   - Open a terminal and log into the PostgreSQL shell:  
+
+     ```bash
+     psql postgres
+     ```
+
+3. **Start the services**
+    - Start the PostgreSQL service:
+
+      ```bash
+        brew services start postgresql
+      ```
```

Comment:
These instructions are still MacOS-specific, especially starting the service. This section needs to make all OS-dependent steps for the install clear.